components/UI/profileCard/profileCard.tsx

```diff
@@ -0,0 +1,180 @@
+import React, {
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useState,
+} from "react";
+import styles from "@styles/dashboard.module.css";
+import CopyIcon from "@components/UI/iconsComponents/icons/copyIcon";
+import { CDNImage } from "@components/cdn/image";
+import { useStarkProfile } from "@starknet-react/core";
+import { minifyAddressFromStrings } from "@utils/stringService";
+import xpIcon from "public/icons/xpBadge.svg";
+
+import useCreationDate from "@hooks/useCreationDate";
+import shareSrc from "public/icons/share.svg";
+import SharePopup from "../menus/sharePopup";
+import theme from "@styles/theme";
+import { Tooltip } from "@mui/material";
+import VerifiedIcon from "../iconsComponents/icons/verifiedIcon";
+import ProfilIcon from "../iconsComponents/icons/profilIcon";
+import Link from "next/link";
+import SocialMediaActions from "../actions/socialmediaActions";
+import { getTweetLink } from "@utils/browserService";
+import { hexToDecimal } from "@utils/feltService";
+import { calculatePercentile } from "@utils/numberService";
+
+const ProfileCard: FunctionComponent<ProfileCard> = ({
+  rankingData,
+  identity,
+  leaderboardData,
+  addressOrDomain,
+  isOwner,
+}) => {
+  const [copied, setCopied] = useState(false);
+  const sinceDate = useCreationDate(identity);
+  const { data: profileData } = useStarkProfile({ address: identity.owner });
+  const [showSharePopup, setShowSharePopup] = useState(false);
+  const [userPercentile, setUserPercentile] = useState("");
+  const [userXp, setUserXp] = useState<number>();
+
+  const copyToClipboard = () => {
```

Comment:
This should be in utils somewhere as we might reuse it (most probably `stringService.ts` then you have to test it (using GPT it will be very fast).

https://www.notion.so/lfg-labs/Jest-Test-GPT-Prompts-3112ecd47c454718b61af2e8ecc5a1d4

Reply:
Are you sure it's done ? I still see the function here :)

Reply:
handling it here - https://github.com/starknet-id/starknet.quest/pull/599/files#diff-b8c7e8bc365baf9a766dc8147a6da3817f92e4ca41712b6e2787dd439505a4c9R43

This function will only handle the copy loader timer function. The util is now `writeToClipboard` which takes in the data and saves it in the clipboard. The tests are added for it too